File: BudgetZeroApp/app/src/main/java/com/example/budgetzeroapp/fragment/savings/SavingsFragment.java
package com.example.budgetzeroapp.fragment.savings;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.budgetzeroapp.R;
import com.example.budgetzeroapp.fragment.DataBaseFragment;
import com.example.budgetzeroapp.tool.adapter.ViewPagerAdapter;
import com.google.android.material.tabs.TabLayout;


public class SavingsFragment extends DataBaseFragment {
    private TabLayout tabLayout;
    private ViewPager viewPager;
    private boolean defaultTab;

    public SavingsFragment() {
        super();
        defaultTab = true;
    }

    public SavingsFragment(boolean tab){
        super();
        defaultTab = tab;
    }

    public static SavingsFragment newInstance(String param1, String param2) {
        return new SavingsFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_savings, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        tabLayout = view.findViewById(R.id.tab_layout);
        viewPager = view.findViewById(R.id.viewPager);
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        adapter.addFragment(new SavingstTabFragment(), "Savings");
        adapter.addFragment(new SavingsDebtsTabFragment(), "Debts");
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
        if(!defaultTab) selectPage(2);

        /**Setting toolbar title**/
        Toolbar toolbar = view.findViewById(R.id.toolbar_savings);
        toolbar.setTitle("Savings");
    }

    public void selectPage(int pageIndex){
        tabLayout.setScrollPosition(pageIndex,0f,true);
        viewPager.setCurrentItem(pageIndex);
    }
}

File: BokuDarjan/app/src/main/java/com/example/bokudarjan/SavingsFragment.kt
package com.example.bokudarjan

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.bokudarjan.dialog.AddSavingDialog
import com.example.bokudarjan.saving.ListAdapterSaving
import com.example.bokudarjan.saving.Saving
import com.example.bokudarjan.saving.SavingViewModel
import kotlinx.android.synthetic.main.fragment_savings.view.*

// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A [Fragment] subclass, used to display the saving section on the main activity of the app.
 * It contains a Recyclerview to display saving projects as well as a button to add one.
 * Use the [ExpensesFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class SavingsFragment : Fragment() {
    // TODO: Rename and change types of parameters
    private var param1: String? = null
    private var param2: String? = null
    private lateinit var savingViewModel : SavingViewModel;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }

        savingViewModel = ViewModelProvider(this).get(SavingViewModel::class.java)
    }

    /**
     * Initializing the fragment (setting up recyclerView, onClickListener, ...)
     */
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        var view =  inflater.inflate(R.layout.fragment_savings, container, false)

        var test = listOf<Saving>(Saving("Acheter une voiture", 1000f), Saving("Maison", 100f));

        var savingAdapter = ListAdapterSaving()
        view.recylclerSavings.adapter = savingAdapter;
        view.recylclerSavings.layoutManager = LinearLayoutManager(requireContext())

        savingViewModel.readAllData.observe(viewLifecycleOwner, Observer { saving ->
            savingAdapter.setData(saving)
        })

        view.expandButton3.setOnClickListener {
            var dialog = AddSavingDialog()
            dialog.show(childFragmentManager, "")
        }

        return view;
    }
}